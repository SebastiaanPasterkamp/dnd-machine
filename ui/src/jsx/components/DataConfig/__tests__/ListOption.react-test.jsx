import React from 'react';
import _ from 'lodash';
import fp from 'lodash/fp';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ListDataActions from 'actions/ListDataActions';
import {
    mockedApi,
    armor_types,
} from '../../../../../tests/__mocks__';

import { ListOption } from '..';
import uuidv4 from '../utils/uuidv4';
jest.mock('../utils/uuidv4');

const mockedId1 = '1';
const mockedId2 = '2';
const mockedUUID1 = 'mocked-uuid-1';

describe('Component: ListOption', () => {
    const fullProps = {
        type: 'list',
        uuid: mockedUUID1,
        path: 'some.path',
        given: ["2", '1 * level'],
        label: 'Something given',
        description: "You're welcome",
        multiple: true,
        hidden: true,
    };

    const listProps = {
        type: 'list',
        uuid: mockedUUID1,
        path: 'proficiencies.armor',
        list: ['armor_types'],
        filter: {
            type: 'shield',
        },
        given: ['shield'],
        label: 'Some armor',
        description: "You're welcome",
    };

    beforeAll(() => {
        fetch.mockImplementation(mockedApi({
            armor_types,
        }));
        ListDataActions.fetchItems('armor_types');
    });

    afterAll(() => fetch.resetMocks());

    beforeEach(() => {
        fp.uniqueId = _.uniqueId = jest.fn();
        _.uniqueId
            .mockReturnValueOnce(mockedId1)
            .mockReturnValueOnce(mockedId2)
            .mockReturnValueOnce('unexpected');
        uuidv4.mockImplementation(
            jest.fn().mockReturnValueOnce(mockedUUID1)
        );
    });

    afterEach(() => jest.clearAllMocks());

    describe('when rendering', () => {
        const setState = jest.fn();

        it('should work with minimum props', () => {
            const tree = renderer.create(
                <ListOption
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('should work with full props', () => {
            const tree = renderer.create(
                <ListOption
                    {...fullProps}
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('should show label/desc when not hidden', () => {
            const tree = renderer.create(
                <ListOption
                    {...fullProps}
                    hidden={false}
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('should show label/desc when cannot be hidden', () => {
            const tree = renderer.create(
                <ListOption
                    {...fullProps}
                    canBeHidden={false}
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('should show more options with a list', () => {
            const tree = renderer.create(
                <ListOption
                    {...listProps}
                    setState={setState}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        })
    });

    it('should emit changes in the label', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <ListOption
                setState={setState}
            />
        );

        wrapper.find('input[type="text"]').at(1).simulate('change', {
            target: { value: fullProps.label }
        });

        expect(setState).toBeCalledWith({
            type: 'list',
            uuid: mockedUUID1,
            label: fullProps.label,
        });
    });

    it('should emit changes in the child component', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <ListOption
                {...fullProps}
                setState={setState}
            />
        );

        wrapper.find('input[type="text"]').at(1).simulate('change', {
            target: { value: "3" }
        });

        expect(setState).toBeCalledWith({
            type: 'list',
            uuid: mockedUUID1,
            given: [ 3, '1 * level' ],
        });
    });

    it('should add items to the list types', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <ListOption
                setState={setState}
            />
        );

        wrapper.find('.base-tag-container button').simulate('click');
        wrapper.find('li[data-value="armor_types"]').simulate('click');

        expect(setState).toBeCalledWith({
            type: 'list',
            uuid: mockedUUID1,
            list: ['armor_types'],
        });
    });

    it('should emit list filters when added', () => {
        const setState = jest.fn();

        const wrapper = mount(
            <ListOption
                list={['armor_types']}
                add={1}
                setState={setState}
            />
        );

        wrapper.find('div[name="add"] button').simulate('click');
        wrapper.find('li[data-value="boolean"]').simulate('click');

        expect(setState).toBeCalledWith({
            type: 'list',
            uuid: mockedUUID1,
            filter: {
                '': true,
            },
        });
    });

    describe('when editing fields that accept formulas', () => {
        let setState, wrapper;

        beforeEach(() => {
            setState = jest.fn();

            wrapper = mount(
                <ListOption
                    list={['armor_types']}
                    setState={setState}
                />
            );
        });

        it('should emit a _formula field', () => {
            wrapper.find('input').at(4).simulate('change', {
                target: { value: "spells.max_cantrips" }
            });

            expect(setState).toBeCalledWith({
                type: 'list',
                uuid: mockedUUID1,
                limit: undefined,
                limit_formula: "spells.max_cantrips",
            });
        });

        it('should convert numeric values without _formula', () => {
            wrapper.find('input').at(4).simulate('change', {
                target: { value: "4" }
            });

            expect(setState).toBeCalledWith({
                type: 'list',
                uuid: mockedUUID1,
                limit: 4,
                limit_formula: undefined,
            });
        });

        it('should blank the fields when the value is removed', () => {
            wrapper.find('input').at(4).simulate('change', {
                target: { value: "" }
            });

            expect(setState).toBeCalledWith({
                type: 'list',
                uuid: mockedUUID1,
                limit: undefined,
                limit_formula: undefined,
            });
        });
    });
});
