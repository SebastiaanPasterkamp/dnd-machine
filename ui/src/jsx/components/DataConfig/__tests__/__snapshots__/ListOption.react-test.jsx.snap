// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ListOption when rendering should show label/desc when cannot be hidden 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    List option
  </legend>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Path
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Path..."
      type="text"
      value="some.path"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Label
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Label..."
      type="text"
      value="Something given"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Description
    </span>
    <div
      className="markdown-textedit small preview"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <textarea
        className="nice-form-control"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Description..."
        rows={8}
        value="You're welcome"
      />
      <span
        className="markdown-textedit__preview"
      >
        <p>
          You're welcome
        </p>
      </span>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Lists
    </span>
    <div
      className="base-tag-container nice-tags-container"
    >
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          disabled={false}
          onClick={[Function]}
        >
          Add...
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={[Function]}
        >
          <li
            className={null}
            data-value="armor"
            onClick={[Function]}
          >
            <a>
              Armor
            </a>
          </li>
          <li
            className={null}
            data-value="armor_types"
            onClick={[Function]}
          >
            <a>
              Armor types
            </a>
          </li>
          <li
            className={null}
            data-value="gear"
            onClick={[Function]}
          >
            <a>
              Gear
            </a>
          </li>
          <li
            className={null}
            data-value="humanoid_types"
            onClick={[Function]}
          >
            <a>
              Humanoid types
            </a>
          </li>
          <li
            className={null}
            data-value="languages"
            onClick={[Function]}
          >
            <a>
              Languages
            </a>
          </li>
          <li
            className={null}
            data-value="monster_types"
            onClick={[Function]}
          >
            <a>
              Monster types
            </a>
          </li>
          <li
            className={null}
            data-value="skills"
            onClick={[Function]}
          >
            <a>
              Skills
            </a>
          </li>
          <li
            className={null}
            data-value="spell"
            onClick={[Function]}
          >
            <a>
              Spells
            </a>
          </li>
          <li
            className={null}
            data-value="statistics"
            onClick={[Function]}
          >
            <a>
              Statistics
            </a>
          </li>
          <li
            className={null}
            data-value="terrain_types"
            onClick={[Function]}
          >
            <a>
              Terrain types
            </a>
          </li>
          <li
            className={null}
            data-value="equipment_types"
            onClick={[Function]}
          >
            <a>
              Tool types
            </a>
          </li>
          <li
            className={null}
            data-value="weapon"
            onClick={[Function]}
          >
            <a>
              Weapons
            </a>
          </li>
          <li
            className={null}
            data-value="weapon_types"
            onClick={[Function]}
          >
            <a>
              Weapon types
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Given
    </span>
    <ul
      className="list-component"
    >
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="2"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="1 * level"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={true}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
    </ul>
  </div>
  <input
    checked={true}
    className="nice-toggle"
    disabled={undefined}
    id="toggle-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="nice-toggle-btn"
    htmlFor="toggle-1"
  >
    Can have duplicates in the same list (multiple)
  </label>
</fieldset>
`;

exports[`Component: ListOption when rendering should show label/desc when not hidden 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    List option
  </legend>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Path
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Path..."
      type="text"
      value="some.path"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Display
    </span>
    <div
      className="nice-dropdown nice-form-control"
    >
      <button
        className="nice-btn"
        disabled={false}
        onClick={[Function]}
      >
        Everything
        <i
          className="icon fa-angle-down"
        />
      </button>
      <ul
        className="dropdown-menu"
        onClick={[Function]}
      >
        <li
          className="info"
          data-value={false}
          onClick={[Function]}
        >
          <a>
            Everything
          </a>
        </li>
        <li
          className={null}
          data-value={true}
          onClick={[Function]}
        >
          <a>
            Nothing
          </a>
        </li>
      </ul>
    </div>
    <span
      className="nice-input-addon"
    >
      Formula
    </span>
    <input
      className="nice-form-control"
      disabled={false}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="character.level < 5"
      type={undefined}
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Label
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Label..."
      type="text"
      value="Something given"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Description
    </span>
    <div
      className="markdown-textedit small preview"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <textarea
        className="nice-form-control"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Description..."
        rows={8}
        value="You're welcome"
      />
      <span
        className="markdown-textedit__preview"
      >
        <p>
          You're welcome
        </p>
      </span>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Lists
    </span>
    <div
      className="base-tag-container nice-tags-container"
    >
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          disabled={false}
          onClick={[Function]}
        >
          Add...
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={[Function]}
        >
          <li
            className={null}
            data-value="armor"
            onClick={[Function]}
          >
            <a>
              Armor
            </a>
          </li>
          <li
            className={null}
            data-value="armor_types"
            onClick={[Function]}
          >
            <a>
              Armor types
            </a>
          </li>
          <li
            className={null}
            data-value="gear"
            onClick={[Function]}
          >
            <a>
              Gear
            </a>
          </li>
          <li
            className={null}
            data-value="humanoid_types"
            onClick={[Function]}
          >
            <a>
              Humanoid types
            </a>
          </li>
          <li
            className={null}
            data-value="languages"
            onClick={[Function]}
          >
            <a>
              Languages
            </a>
          </li>
          <li
            className={null}
            data-value="monster_types"
            onClick={[Function]}
          >
            <a>
              Monster types
            </a>
          </li>
          <li
            className={null}
            data-value="skills"
            onClick={[Function]}
          >
            <a>
              Skills
            </a>
          </li>
          <li
            className={null}
            data-value="spell"
            onClick={[Function]}
          >
            <a>
              Spells
            </a>
          </li>
          <li
            className={null}
            data-value="statistics"
            onClick={[Function]}
          >
            <a>
              Statistics
            </a>
          </li>
          <li
            className={null}
            data-value="terrain_types"
            onClick={[Function]}
          >
            <a>
              Terrain types
            </a>
          </li>
          <li
            className={null}
            data-value="equipment_types"
            onClick={[Function]}
          >
            <a>
              Tool types
            </a>
          </li>
          <li
            className={null}
            data-value="weapon"
            onClick={[Function]}
          >
            <a>
              Weapons
            </a>
          </li>
          <li
            className={null}
            data-value="weapon_types"
            onClick={[Function]}
          >
            <a>
              Weapon types
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Given
    </span>
    <ul
      className="list-component"
    >
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="2"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="1 * level"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={true}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
    </ul>
  </div>
  <input
    checked={true}
    className="nice-toggle"
    disabled={undefined}
    id="toggle-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="nice-toggle-btn"
    htmlFor="toggle-1"
  >
    Can have duplicates in the same list (multiple)
  </label>
</fieldset>
`;

exports[`Component: ListOption when rendering should show more options with a list 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    List option
  </legend>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Path
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Path..."
      type="text"
      value="proficiencies.armor"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Display
    </span>
    <div
      className="nice-dropdown nice-form-control"
    >
      <button
        className="nice-btn"
        disabled={false}
        onClick={[Function]}
      >
        Everything
        <i
          className="icon fa-angle-down"
        />
      </button>
      <ul
        className="dropdown-menu"
        onClick={[Function]}
      >
        <li
          className="info"
          data-value={false}
          onClick={[Function]}
        >
          <a>
            Everything
          </a>
        </li>
        <li
          className={null}
          data-value={true}
          onClick={[Function]}
        >
          <a>
            Nothing
          </a>
        </li>
      </ul>
    </div>
    <span
      className="nice-input-addon"
    >
      Formula
    </span>
    <input
      className="nice-form-control"
      disabled={false}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="character.level < 5"
      type={undefined}
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Label
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Label..."
      type="text"
      value="Some armor"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Description
    </span>
    <div
      className="markdown-textedit small preview"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <textarea
        className="nice-form-control"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Description..."
        rows={8}
        value="You're welcome"
      />
      <span
        className="markdown-textedit__preview"
      >
        <p>
          You're welcome
        </p>
      </span>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Lists
    </span>
    <div
      className="base-tag-container nice-tags-container"
    >
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          disabled={false}
          onClick={[Function]}
        >
          Add...
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={[Function]}
        >
          <li
            className={null}
            data-value="armor"
            onClick={[Function]}
          >
            <a>
              Armor
            </a>
          </li>
          <li
            className={null}
            data-value="gear"
            onClick={[Function]}
          >
            <a>
              Gear
            </a>
          </li>
          <li
            className={null}
            data-value="humanoid_types"
            onClick={[Function]}
          >
            <a>
              Humanoid types
            </a>
          </li>
          <li
            className={null}
            data-value="languages"
            onClick={[Function]}
          >
            <a>
              Languages
            </a>
          </li>
          <li
            className={null}
            data-value="monster_types"
            onClick={[Function]}
          >
            <a>
              Monster types
            </a>
          </li>
          <li
            className={null}
            data-value="skills"
            onClick={[Function]}
          >
            <a>
              Skills
            </a>
          </li>
          <li
            className={null}
            data-value="spell"
            onClick={[Function]}
          >
            <a>
              Spells
            </a>
          </li>
          <li
            className={null}
            data-value="statistics"
            onClick={[Function]}
          >
            <a>
              Statistics
            </a>
          </li>
          <li
            className={null}
            data-value="terrain_types"
            onClick={[Function]}
          >
            <a>
              Terrain types
            </a>
          </li>
          <li
            className={null}
            data-value="equipment_types"
            onClick={[Function]}
          >
            <a>
              Tool types
            </a>
          </li>
          <li
            className={null}
            data-value="weapon"
            onClick={[Function]}
          >
            <a>
              Weapons
            </a>
          </li>
          <li
            className={null}
            data-value="weapon_types"
            onClick={[Function]}
          >
            <a>
              Weapon types
            </a>
          </li>
        </ul>
      </div>
      <div
        className="nice-tag"
      >
        <span
          className="nice-tag-label"
        >
          Armor types
        </span>
        <button
          className="nice-tag-btn"
          onClick={[Function]}
        >
          <i
            className="icon fa-trash-o"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Add
    </span>
    <input
      className="nice-form-control"
      disabled={false}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Add..."
      type={undefined}
      value=""
    />
    <span
      className="nice-input-addon"
    >
      Limit
    </span>
    <input
      className="nice-form-control"
      disabled={false}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Limit..."
      type={undefined}
      value=""
    />
    <span
      className="nice-input-addon"
    >
      Replace
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Replace..."
      type={undefined}
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Given
    </span>
    <div
      className="base-tag-container nice-tags-container"
    >
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          disabled={false}
          onClick={[Function]}
        >
          Add...
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={[Function]}
        >
          <input
            className="nice-form-control"
            data-name="filter"
            disabled={undefined}
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={null}
            placeholder="Filter..."
            type="text"
            value=""
          />
          <li
            className={null}
            data-value="light"
            onClick={[Function]}
          >
            <a>
              Light Armor
            </a>
          </li>
          <li
            className={null}
            data-value="medium"
            onClick={[Function]}
          >
            <a>
              Medium Armor
            </a>
          </li>
          <li
            className={null}
            data-value="heavy"
            onClick={[Function]}
          >
            <a>
              Heavy Armor
            </a>
          </li>
        </ul>
      </div>
      <div
        className="nice-tag"
      >
        <span
          className="nice-tag-label"
        >
          Shields
        </span>
        <button
          className="nice-tag-btn"
          onClick={[Function]}
        >
          <i
            className="icon fa-trash-o"
          />
        </button>
      </div>
    </div>
  </div>
  <input
    checked={false}
    className="nice-toggle"
    disabled={undefined}
    id="toggle-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="nice-toggle-btn"
    htmlFor="toggle-1"
  >
    Can have duplicates in the same list (multiple)
  </label>
</fieldset>
`;

exports[`Component: ListOption when rendering should work with full props 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    List option
  </legend>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Path
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Path..."
      type="text"
      value="some.path"
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Display
    </span>
    <div
      className="nice-dropdown nice-form-control"
    >
      <button
        className="nice-btn"
        disabled={false}
        onClick={[Function]}
      >
        Nothing
        <i
          className="icon fa-angle-down"
        />
      </button>
      <ul
        className="dropdown-menu"
        onClick={[Function]}
      >
        <li
          className={null}
          data-value={false}
          onClick={[Function]}
        >
          <a>
            Everything
          </a>
        </li>
        <li
          className="info"
          data-value={true}
          onClick={[Function]}
        >
          <a>
            Nothing
          </a>
        </li>
      </ul>
    </div>
    <span
      className="nice-input-addon"
    >
      Formula
    </span>
    <input
      className="nice-form-control"
      disabled={true}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="character.level < 5"
      type={undefined}
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Given
    </span>
    <ul
      className="list-component"
    >
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="2"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          placeholder={undefined}
          type="text"
          value="1 * level"
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={false}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={true}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
    </ul>
  </div>
  <input
    checked={true}
    className="nice-toggle"
    disabled={undefined}
    id="toggle-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="nice-toggle-btn"
    htmlFor="toggle-1"
  >
    Can have duplicates in the same list (multiple)
  </label>
</fieldset>
`;

exports[`Component: ListOption when rendering should work with minimum props 1`] = `
<fieldset
  className="nice-form-group"
>
  <legend>
    List option
  </legend>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Path
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Path..."
      type="text"
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Display
    </span>
    <div
      className="nice-dropdown nice-form-control"
    >
      <button
        className="nice-btn"
        disabled={false}
        onClick={[Function]}
      >
        Everything
        <i
          className="icon fa-angle-down"
        />
      </button>
      <ul
        className="dropdown-menu"
        onClick={[Function]}
      >
        <li
          className="info"
          data-value={false}
          onClick={[Function]}
        >
          <a>
            Everything
          </a>
        </li>
        <li
          className={null}
          data-value={true}
          onClick={[Function]}
        >
          <a>
            Nothing
          </a>
        </li>
      </ul>
    </div>
    <span
      className="nice-input-addon"
    >
      Formula
    </span>
    <input
      className="nice-form-control"
      disabled={false}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="character.level < 5"
      type={undefined}
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Label
    </span>
    <input
      className="nice-form-control"
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={null}
      placeholder="Label..."
      type="text"
      value=""
    />
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Description
    </span>
    <div
      className="markdown-textedit small preview"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span
        className="markdown-textedit__placeholder"
      >
        Description...
      </span>
      <textarea
        className="nice-form-control"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Description..."
        rows={8}
        value=""
      />
      <span
        className="markdown-textedit__preview"
      />
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Lists
    </span>
    <div
      className="base-tag-container nice-tags-container"
    >
      <div
        className="nice-dropdown nice-form-control"
      >
        <button
          className="nice-btn"
          disabled={false}
          onClick={[Function]}
        >
          Add...
          <i
            className="icon fa-angle-down"
          />
        </button>
        <ul
          className="dropdown-menu"
          onClick={[Function]}
        >
          <li
            className={null}
            data-value="armor"
            onClick={[Function]}
          >
            <a>
              Armor
            </a>
          </li>
          <li
            className={null}
            data-value="armor_types"
            onClick={[Function]}
          >
            <a>
              Armor types
            </a>
          </li>
          <li
            className={null}
            data-value="gear"
            onClick={[Function]}
          >
            <a>
              Gear
            </a>
          </li>
          <li
            className={null}
            data-value="humanoid_types"
            onClick={[Function]}
          >
            <a>
              Humanoid types
            </a>
          </li>
          <li
            className={null}
            data-value="languages"
            onClick={[Function]}
          >
            <a>
              Languages
            </a>
          </li>
          <li
            className={null}
            data-value="monster_types"
            onClick={[Function]}
          >
            <a>
              Monster types
            </a>
          </li>
          <li
            className={null}
            data-value="skills"
            onClick={[Function]}
          >
            <a>
              Skills
            </a>
          </li>
          <li
            className={null}
            data-value="spell"
            onClick={[Function]}
          >
            <a>
              Spells
            </a>
          </li>
          <li
            className={null}
            data-value="statistics"
            onClick={[Function]}
          >
            <a>
              Statistics
            </a>
          </li>
          <li
            className={null}
            data-value="terrain_types"
            onClick={[Function]}
          >
            <a>
              Terrain types
            </a>
          </li>
          <li
            className={null}
            data-value="equipment_types"
            onClick={[Function]}
          >
            <a>
              Tool types
            </a>
          </li>
          <li
            className={null}
            data-value="weapon"
            onClick={[Function]}
          >
            <a>
              Weapons
            </a>
          </li>
          <li
            className={null}
            data-value="weapon_types"
            onClick={[Function]}
          >
            <a>
              Weapon types
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="nice-control-group"
  >
    <span
      className="nice-input-addon"
    >
      Given
    </span>
    <ul
      className="list-component"
    >
      <li>
        <input
          className="nice-form-control"
          disabled={undefined}
          onChange={[Function]}
          onKeyPress={null}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        <button
          className="nice-btn list-component__delete warning icon fa-minus"
          disabled={true}
          name="del"
          onClick={[Function]}
        >
          ​
        </button>
      </li>
    </ul>
  </div>
  <input
    checked={false}
    className="nice-toggle"
    disabled={undefined}
    id="toggle-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="nice-toggle-btn"
    htmlFor="toggle-1"
  >
    Can have duplicates in the same list (multiple)
  </label>
</fieldset>
`;
